### PR TITLE
Add deriving metrics from logs use case to Data Prepper

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Add deriving metrics from logs use case to Data Prepper

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6429

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@dlvenable This use case PR is ready for your review at availability. Thank you, Melissa

**Comment by vagimeli** [comment]
@dlvenable At your availability, will you review this use case? It's one of the two last use cases we're transferring from OpenSearch Ingestion. Thank you, Melissa

**Comment by vagimeli** [comment]
> @dlvenable At your availability, will you review this use case? It's one of the two last use cases we're transferring from OpenSearch Ingestion. Thank you, Melissa

Hi @dlvenable, Is it possible to review this PR by end of week? We need to get this done to address feedback in the benchmark study. Thank you.

**Comment by vagimeli** [comment]
@dlvenable Following up to get your tech review on this use case carryover from OpenSearch Service Ingestion to Data Prepper. Once we have your approval, I can move this forward to editorial and get it published.

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (dlvenable) [inline _data-prepper/common-use-cases/metrics-logs.md:10]:
```suggestion
You can use Data Prepper to derive metrics from logs. The following example pipeline receives incoming logs using the [`http` source plugin]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/http-source) and the [`grok` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/grok/). It then uses the [`aggregate` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/aggregate/) to extract the metric bytes aggregated over a 30-second window and derives histograms from the results.
```

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (dlvenable) [inline _data-prepper/common-use-cases/metrics-logs.md:14]:
```suggestion
- `apache-log-pipeline-with-metrics` -– Receives logs through an HTTP client like FluentBit, uses `grok` to extract important values from the logs by matching the value in the `log` key against the [Apache Common Log Format](https://httpd.apache.org/docs/2.4/logs.html#accesslog), and then forwards the grokked logs to both the `log-to-metrics-pipeline` pipeline and to an OpenSearch index named `logs`.
```

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thank you @vagimeli !

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (natebower) [inline _data-prepper/common-use-cases/metrics-logs.md:10]:
```suggestion
You can use Data Prepper to derive metrics from logs. The following example pipeline receives incoming logs using the [`http` source plugin]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/http-source) and the [`grok` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/grok/). It then uses the [`aggregate` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/aggregate/) to extract the metric bytes aggregated during a 30-second window and derives histograms from the results.
```

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (natebower) [inline _data-prepper/common-use-cases/metrics-logs.md:14]:
```suggestion
- `apache-log-pipeline-with-metrics` -- Receives logs through an HTTP client like FluentBit, uses `grok` to extract important values from the logs by matching the value in the `log` key against the [Apache Common Log Format](https://httpd.apache.org/docs/2.4/logs.html#accesslog), and then forwards the grokked logs to both the `log-to-metrics-pipeline` pipeline and to an OpenSearch index named `logs`.
```

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (natebower) [inline _data-prepper/common-use-cases/metrics-logs.md:12]: "main" or "primary" instead of "overall"?

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (natebower) [inline _data-prepper/common-use-cases/metrics-logs.md:16]:
```suggestion
- `log-to-metrics-pipeline` -- Receives the grokked logs from the `apache-log-pipeline-with-metrics` pipeline, aggregates the logs, and derives histogram metrics of `bytes` based on the values in the `clientip` and `request` keys. Finally, it sends the histogram metrics to an OpenSearch index named `histogram_metrics`.
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

**Comment by vagimeli** [comment]
@dlvenable Thanks for your review. I've accepted your changes. Do we have any other outstanding items to address before merging?

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (dlvenable) [inline _data-prepper/common-use-cases/metrics-logs.md:10]: This writes data to two indexes - one with un-aggregated events and the other derived metrics.

Can we clarify somewhere? You mention below how we use two pipelines. But, maybe we can make this end result more explicit.

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (vagimeli) [inline _data-prepper/common-use-cases/metrics-logs.md:14]: @dlvenable Please review the text added at lines 14--27 and let me know if these additions address your feedback and are accurate. Thank you.

File: _data-prepper/common-use-cases/metrics-logs.md
Comment (vagimeli) [inline _data-prepper/common-use-cases/metrics-logs.md:21]:
```suggestion
- `apache-log-pipeline-with-metrics`: Receives logs through an HTTP client like FluentBit, using `grok` to extract important values from the logs by matching the value in the log key against the [Apache Common Log Format](https://httpd.apache.org/docs/2.4/logs.html#accesslog). It then forwards the grokked logs to two destinations:
```
```suggestion
- `apache-log-pipeline-with-metrics`: Receives logs through an HTTP client like FluentBit, uses `grok` to extract important values from the logs by matching the value in the log key against the [Apache Common Log Format](https://httpd.apache.org/docs/2.4/logs.html#accesslog). It then forwards the grokked logs to two destinations:
```
```suggestion
- `apache-log-pipeline-with-metrics`: Receives logs through an HTTP client like FluentBit, uses `grok` to extract important values from the logs by matching the value in the log key against the [Apache Common Log Format](https://httpd.apache.org/docs/2.4/logs.html#accesslog). It then forwards the grokked logs to two destinations:
```

**Review by dlvenable — APPROVED** [review]

**Pull request merged**
